src/main/java/seedu/command/AddCommand.java

```diff
@@ -0,0 +1,226 @@
+package seedu.command;
+
+import seedu.exceptions.InventraException;
+import seedu.exceptions.InventraInvalidFlagException;
+import seedu.exceptions.InventraInvalidTypeException;
+import seedu.exceptions.InventraMissingFieldsException;
+import seedu.exceptions.InventraInvalidRecordCountException;
+import seedu.model.Inventory;
+import seedu.storage.Csv;
+import seedu.ui.Ui;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+import java.util.logging.Logger;
+
+public class AddCommand {
+    private static final Logger LOGGER = Logger.getLogger(AddCommand.class.getName());
+
+    private final Inventory inventory;
+    private final Ui ui;
+    private final Csv csv;
+
+    public AddCommand(Inventory inventory, Ui ui, Csv csv) {
+        assert inventory != null : "Inventory should not be null";
+        assert ui != null : "Ui should not be null";
+        assert csv != null : "Csv should not be null";
+
+        this.inventory = inventory;
+        this.ui = ui;
+        this.csv = csv;
+    }
+
+    public void execute(String[] args) throws InventraException {
+        assert args != null && args.length >= 2 : "Arguments should contain at least a command and a flag";
+
+        String flag = args[1];
+        assert flag != null && !flag.isEmpty() : "Flag should not be null or empty";
+
+        switch (flag) {
+        case "-h":
```

Comment:
Have these string cases be stored as a constant string; avoid magic strings / numbners.
This is apparent throughout the codebase.